user_crud.c

```diff
@@ -0,0 +1,268 @@
+        fclose(file);
+        return;
+    }
+    targetId = convert_to_int(temp);
+
+    FILE *tempFile = fopen("temp.txt", "w");
+    if (tempFile == NULL) {
+        printf("Error creating temporary file.\n");
+        fclose(file);
+        return;
+    }
+
+    struct User u;
+    while (fscanf(file, "%d,%[^,],%d", &u.id, u.name, &u.age) == 3) {
+        if (u.id == targetId) {
+            found = 1;
+
+            printf("Enter new name: ");
+            scanf(" %[^\n]s", u.name);
+            if (!is_valid_name(u.name)) {
+                printf("Invalid name. Only alphabetic characters and spaces are allowed.\n");
+                fclose(file);
+                fclose(tempFile);
+                remove("temp.txt");
+                return;
+            }
+
+            printf("Enter new age: ");
+            scanf("%s", temp);
+            if (!is_valid_positive_int(temp)) {
+                printf("Invalid age. Please enter a positive integer.\n");
+                fclose(file);
+                fclose(tempFile);
+                remove("temp.txt");
+                return;
+            }
+            u.age = convert_to_int(temp);
+        }
+        fprintf(tempFile, "%d,%s,%d\n", u.id, u.name, u.age);
+    }
+
+    fclose(file);
+    fclose(tempFile);
+
+    remove("users.txt");
+    rename("temp.txt", "users.txt");
+
+    if (found) {
+        printf("User updated successfully.\n");
+    } else {
+        printf("User not found.\n");
+    }
+}
+
+void delete_user() {
+    FILE *file = fopen("users.txt", "r");
+    if (file == NULL) {
+        printf("No users found.\n");
+        return;
+    }
+
+    int targetId, found = 0;
+    char temp[100];
+
+    printf("Enter the User ID to delete: ");
+    scanf("%s", temp);
+    if (!is_valid_positive_int(temp)) {
+        printf("Invalid ID. Please enter a positive integer.\n");
+        fclose(file);
+        return;
+    }
+    targetId = convert_to_int(temp);
+
+    FILE *tempFile = fopen("temp.txt", "w");
+    if (tempFile == NULL) {
+        printf("Error creating temporary file.\n");
+        fclose(file);
+        return;
+    }
+
+    struct User u;
+    while (fscanf(file, "%d,%[^,],%d", &u.id, u.name, &u.age) == 3) {
+        if (u.id == targetId) {
+            found = 1;
+            continue;
+        }
+        fprintf(tempFile, "%d,%s,%d\n", u.id, u.name, u.age);
+    }
+
+    fclose(file);
+    fclose(tempFile);
+
+    remove("users.txt");
+    rename("temp.txt", "users.txt");
+
+    if (found) {
+        printf("User deleted successfully.\n");
+    } else {
+        printf("User not found.\n");
+    }
+}
+
+int main() {
+    int choice;
+    do {
+        printf("\nUser Management System\n");
+        printf("1. Add User\n");
+        printf("2. Display Users\n");
+        printf("3. Update User\n");
+        printf("4. Delete User\n");
+        printf("5. Exit\n");
+        printf("Enter your choice: ");
+        scanf("%d", &choice);
```

Comment:
Edge case needs to be handled. If choice is inputed other then an integer like 'a' then it is going to infinte loop.